Feature/testing w rtl

NewIncome opened the pull request:
Made required tests according to Microverse's Math-Magicians Testing Project

NewIncome committed:
Add the dev dependencies needed to work with Jest testing and
the react-tesnting-library

- Also add the scripts to test with jest
- And add some configurations for eslint

NewIncome committed:
Add files for the corresponding components to be tested

NewIncome committed:
Finished unit tests on calculate.js, react tests not working yet

- Tried installing babel and other components and other config
files and no luck yet for React component tests

NewIncome committed:
Testing the App.js isn not working, other tests are working ok

NewIncome committed:
Make neccesary modifications for tests to pass

NewIncome committed:
Fix linter errors for babel.config

wathigo reviewed (changes requested):
## Changes Required :o: 
Great job adding tests.
Let's make some changes before we move on.

### REQUIRED
* Use jest snapshots to make sure that all the elements in the Home, Calculator, and Quote pages have been displayed. 
![Screenshot from 2021-01-16 06-57-22](https://user-images.githubusercontent.com/44072711/104811281-6a188680-57c8-11eb-98ec-f4f05600a313.png)
* Include jest in devDependencies by running `npm install --save-dev jest`.

**After making the changes, submit another request for review.**

**Happy Coding :computer: 
Simon Wathigo**

NewIncome commented:
> ## Changes Required o
> 
> Great job adding tests.
> Let's make some changes before we move on.
> ### REQUIRED
> 
>     * Use jest snapshots to make sure that all the elements in the Home, Calculator, and Quote pages have been displayed.
>       ![Screenshot from 2021-01-16 06-57-22](https://user-images.githubusercontent.com/44072711/104811281-6a188680-57c8-11eb-98ec-f4f05600a313.png)
> 
>     * Include jest in devDependencies by running `npm install --save-dev jest`.
> 
> 
> **After making the changes, submit another request for review.**
> 
> **Happy Coding computer Simon Wathigo**

Hi @wathigo 
Please need some help.
I've been getting an error and can't manage to solve it. Regarding React tests.
I posted the seek for help in the Full-time student's Slack channel.

By the way, Jest was already in my devDependencies but anyway I ran that line to install it again.

NewIncome committed:
Make many modifications to solve the test of the calculatorPage
Not entirely working yet

- Include the renderer for snapshot testing
- Include prettier, needed for the renderer
- Snapshot tests woking for Home and Quote pages
- Made a separate file to test the calculatorPage in parts
- Include some React imports, which may be the problem Jest is
having (so far only on Display.js and that did fix one error)

NewIncome committed:
Fix linter errors, Fix Jest error, Add a last test for calcPage

- Add the React import (also) to Button.js
- Remove unused App.test.js
- Add filter for stylelint to not check a 'prettier' directory

bellom commented on src/components/App.js:
```diff
@@ -31,22 +30,27 @@ class App extends React.Component {
+        next={next}
+        handleClick={this.handleClick}
+      />
+      // <section className="section1">
+      //   <p>Let&apos;s do some math!</p>
+      //   <div id="calcContainer">
+      //     <div id="cBody">
+      //       <div className="calcTop">
+      //         <Display result={
+      //           (total === null && next === null) ? '0' : next === null ? total : next
+      //           }
+      //         />
+      //       </div>
+      //       <div className="calcBtm">
+      //         <ButtonPanel onClick={this.handleClick} />
+      //       </div>
+      //     </div>
+      //   </div>
+      // </section>
```
It’s better to delete commented code. If you need to review a previous version of your code, use git history. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

**_Kindly remove before merging._**

bellom reviewed (approved):
## Hi @NewIncome  :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next project :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

NewIncome committed:
Delete commented code
